### PR TITLE
Add support for SSLKEYLOGFILE in dapf

### PR DESCRIPTION
**Pull request by mendess**

### CODE REVIEW

File: crates/dapf/src/acceptance/mod.rs [hunk 1]
```diff
@@ -166,6 +166,19 @@ pub enum HttpClient {
 
 impl HttpClient {
     fn reused(leader_tls_identity: Option<Identity>) -> Self {
```
Comment (cjpatton) [inline crates/dapf/src/acceptance/mod.rs:168]: SSLKEYLOGFILE should not be enabled by default, as it is not secure. The user should have to set it with a flag.

**Review by cjpatton — CHANGES_REQUESTED** [review]

File: crates/dapf/src/acceptance/mod.rs (hunk 1 above)
Comment (mendess) [reply crates/dapf/src/acceptance/mod.rs:168]: even if it's only used from a dapf? I guess we run dapf on the edge. I'll add that flag

File: crates/dapf/src/http_client.rs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    inner: HttpClientInner,
+}
+
+#[allow(clippy::large_enum_variant)]
+enum HttpClientInner {
+    NoReuse {
+        identity: Option<Identity>,
+        ssl_key_log_file_enabled_tls: Option<rustls::ClientConfig>,
+    },
+    Reuse(Client),
```
Comment (cjpatton) [inline crates/dapf/src/http_client.rs:28]: A comment here explaining what we mean by "reuse" would be helpful.

File: crates/dapf/src/http_client.rs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+fn load_identity() -> anyhow::Result<Option<Identity>> {
+    // LEADER_ variables left here for backwards compatibility
+    const TLS_CLIENT_CERT_VAR: [&str; 2] = ["LEADER_TLS_CLIENT_CERT", "TLS_CLIENT_CERT"];
+    const TLS_CLIENT_KEY_VAR: [&str; 2] = ["LEADER_TLS_CLIENT_KEY", "TLS_CLIENT_KEY"];
+
+    let leader_tls_client_cert = TLS_CLIENT_CERT_VAR
+        .iter()
+        .find_map(|key| env::var(key).ok());
```
Comment (cjpatton) [inline crates/dapf/src/http_client.rs:38]: nit: Probably better to check for the non-legacy variable first, i.e., "TLS_CLIENT_CERT" should be preferred over "LEADER_TLS_CLIENT_CERT" if both exist

File: crates/dapf/src/http_client.rs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        http_client_builder = http_client_builder.identity(identity);
+    }
+
+    http_client_builder
+        .build()
+        .expect("failed to build http client")
+}
+
+impl HttpClient {
+    pub fn reused(enable_ssl_key_log_file: bool) -> anyhow::Result<Self> {
```
Comment (cjpatton) [inline crates/dapf/src/http_client.rs:99]: ?
```suggestion
    pub fn reuse(enable_ssl_key_log_file: bool) -> anyhow::Result<Self> {
```

**Review by cjpatton — APPROVED** [review]

**Review by cjpatton — APPROVED** [review]

**Pull request merged**